pkg/services/dashboards/database/database.go

```diff
@@ -65,6 +71,54 @@ func ProvideDashboardStore(sqlStore db.DB, cfg *setting.Cfg, features featuremgm
 	return s, nil
 }
 
+func (d *dashboardStore) DBMigration(db db.DB) {
+	ctx := context.Background()
+	var dashboards = make([]*dashboards.Dashboard, 0)
+
+	err := db.WithDbSession(ctx, func(sess *sqlstore.DBSession) error {
+		var err error
+		migTable := "migration_log"
+		mig := migrator.MigrationLog{
+			MigrationID: "Add DB migration for panel_titles",
+			Success:     true,
+		}
+
+		if exists, err := sess.Table(migTable).Get(&mig); exists || (err != nil) {
+			d.log.Info("DB migration on dashboard store start skipped")
+			return err
+		}
+
+		err = sess.Table("dashboard").Find(&dashboards)
+		if err != nil {
+			return err
+		}
+		for _, d := range dashboards {
+			_, err = sess.Exec("UPDATE dashboard SET panel_titles = ? WHERE id = ?", getPanelTitles(d), d.ID)
+			if err != nil {
+				return err
+			}
+		}
+
+		mig.Timestamp = time.Now()
+		_, err = sess.Table(migTable).Insert(&mig)
+		return err
+	})
+	if err != nil {
+		d.log.Error("DB migration on dashboard store start failed:", err)
+	}
+}
+
+func getPanelTitles(dash *dashboards.Dashboard) string {
+	titles := make([]string, 0)
+	panels := dash.Data.Get("panels").MustArray()
+	for _, p := range panels {
+		panel := simplejson.NewFromAny(p)
+		titles = append(titles, panel.Get("title").MustString())
+	}
+	const sep = " ,,, "
```

Comment:
what will happen if a panel title contains this separator?

Reply:
We're going to strip the separator from the title (haven't done this yet) so the panel title won't include the separator when it's stored in the database. Also this isn't the final separator.